### ISSUE TITLE
Add solar-and-wind-potentials as a submodule to reduce code duplication

### ISSUE THREAD

**Issue by timtroendle** [calliope-project/euro-calliope#110]
There is code duplication in the two `libs` and there are rules duplicated between the two workflows. By adding `solar-and-wind-potentials` as a subworkflow, we can get rid of the code duplication.

**Comment by wwwuFei** [comment]
test

**Comment by timtroendle** [comment]
This solves some of the issue #77.

**Comment by timtroendle** [comment]
@wwwuFei and I looked into this last week. The libs do not have any overlap at this point. Rules that build shapes, and rules that generate capacity factor timeseries on these shapes do overlap (in fact they are out of sync between the two workflows).

We did not find a simple and clean way to get rid of only this small overlap without linking both repositories stronger than intended. When `solar-and-wind-potentials` is imported as a subworkflow or module, Snakemake requires the entire configuration to be present. This will require us to have configuration values in `euro-calliope` that have no meaning or use. This approach also means to add a Git submodule to euro-calliope. Git submodules are usually not easy to handle and this would therefore increase the complexity of euro-calliope.

Before starting with some half-thought-through solutions, I think we should work on our long-term plan with these two repositories first (over in #77). @brynpickering, is this fine for you? If yes, I will close this issue in favour of #77.
